flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/maintenance/api/ZkLockFactory.java

```diff
@@ -126,29 +159,46 @@ public void close() throws IOException {
       if (client != null) {
         client.close();
       }
+      isOpen = false;
     }
   }
 
   /** Zookeeper lock implementation */
   private static class ZkLock implements Lock {
     private final SharedCount sharedCount;
+    private final String lockId;
+    private final String lockType;
+    private final String lockPath;
+
+    private static final int LOCKED = 1;
+    private static final int UNLOCKED = 0;
 
-    private ZkLock(SharedCount sharedCount) {
+    private ZkLock(String lockId, String lockType, String lockPath, SharedCount sharedCount) {
+      this.lockId = lockId;
+      this.lockType = lockType;
+      this.lockPath = lockPath;
       this.sharedCount = sharedCount;
     }
 
     @Override
     public boolean tryLock() {
       VersionedValue<Integer> versionedValue = sharedCount.getVersionedValue();
       if (isHeld(versionedValue)) {
-        LOG.debug("Lock is already held for {}", this);
+        LOG.debug(
+            "Lock is already held for lockId: {}, type: {}, path: {}.", lockId, lockType, lockPath);
         return false;
       }
 
       try {
-        return sharedCount.trySetCount(versionedValue, LOCKED);
+        boolean acquired = sharedCount.trySetCount(versionedValue, LOCKED);
+        if (!acquired) {
+          LOG.warn(
```

Comment:
I don't think we would like to have a `warn` level log for failing to acquire the lock. Max info

Reply:
And even `INFO` can cause a flood of log messages. Maybe `DEBUG` would be better

Reply:
Thank you for your suggestions, I will make the changes.